pkg/cmd/server/origin/master.go

```diff
@@ -382,7 +382,13 @@ func (c *MasterConfig) GetRestStorage() map[string]rest.Storage {
 	checkStorageErr(err)
 	buildConfigRegistry := buildconfigregistry.NewRegistry(buildConfigStorage)
 
-	deployConfigStorage, deployConfigStatusStorage, deployConfigScaleStorage, err := deployconfigetcd.NewREST(c.RESTOptionsGetter, c.DeploymentConfigScaleClient())
+	deployConfigStorage, deployConfigStatusStorage, deployConfigScaleStorage, dcInstantiateStorage, err := deployconfigetcd.NewREST(
```

Comment:
This is starting to look a little silly.  Perhaps we should have separate constructors that accept bits they need as args.